extension/storage/dbstorage/extension_test.go

```diff
@@ -112,6 +123,48 @@ func newTestExtension(t *testing.T) storage.Extension {
 	return se
 }
 
+func newPostgresTestExtension(t *testing.T) storage.Extension {
+	req := testcontainers.GenericContainerRequest{
+		ContainerRequest: testcontainers.ContainerRequest{
+			Image: "postgres:14",
+			HostConfigModifier: func(config *ctypes.HostConfig) {
+				ports := nat.PortMap{}
+				ports[nat.Port("5432")] = []nat.PortBinding{
+					{HostPort: "5432"},
+				}
+				config.PortBindings = ports
+			},
+			Env: map[string]string{
+				"POSTGRES_PASSWORD": "passwd",
+				"POSTGRES_USER":     "root",
+				"POSTGRES_DB":       "db",
+			},
+			WaitingFor: wait.ForListeningPort("5432"),
+		},
+		Started: true,
+	}
+
+	ctr, err := testcontainers.GenericContainer(context.Background(), req)
+	require.NoError(t, err)
+	port, err := ctr.MappedPort(context.Background(), "5432")
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		require.NoError(t, ctr.Terminate(context.Background()))
+	})
+	f := NewFactory()
+	cfg := f.CreateDefaultConfig().(*Config)
+	cfg.DriverName = "pgx"
+	cfg.DataSource = fmt.Sprintf("host=%s port=%s user=%s password=%s dbname=%s sslmode=disable", "127.0.0.1", port.Port(), "root", "passwd", "db")
+
+	extension, err := f.CreateExtension(context.Background(), extensiontest.NewNopSettings(), cfg)
```

Comment:
It'd be nice to do some set/get on the storage after initialization

Reply:
yes, we do that in the test, this code sets up the storage client.